tracer/build/_build/NukeExtensions/DotNetSettingsExtensions.cs

```diff
@@ -126,4 +170,43 @@ public static T SetDotnetPath<T>(this T settings, MSBuildTargetPlatform platform
 
         return settings.SetProcessToolPath(dotnetPath);
     }
+
+    public static DotNetBuildSettings SetTargetFramework(this DotNetBuildSettings settings, TargetFramework framework)
+    {
+        return framework is null
+                   ? settings
+                   : settings.SetFramework(framework);
+    }
+
+    public static DotNetTestSettings SetTargetFramework(this DotNetTestSettings settings, TargetFramework framework)
+    {
+        return framework is null
+                   ? settings
+                   : settings.SetFramework(framework);
+    }
```

Comment:
Not a fan of these extensions personally, and we can use the builtin alternative `When()` I think.  I would be in favour of introducing a shorthand `When` extension though! e.g.

```csharp
public static T When<T>(this T settings, string value, Configure<T> configurator)
        where T : ToolSettings
        => settings.When(!string.IsNullOrEmpty(value), configurator);
```

Reply:
I'm totally agree with you.
I'm not a fun of extensions methods in general, and "1 line" methods in general.
Will use your suggestion, really like it.

I've just sticked to another extension method in the same class:
https://github.com/DataDog/dd-trace-dotnet/blob/cecac4ffd3825a1c876e8e98eaa513b80d523744/tracer/build/_build/NukeExtensions/DotNetSettingsExtensions.cs#L27